crates/swc_ecma_transforms_typescript/src/typescript.rs

```diff
@@ -0,0 +1,98 @@
+use serde::{Deserialize, Serialize};
+use swc_ecma_ast::*;
+use swc_ecma_visit::{as_folder, Fold, VisitMut, VisitMutWith, VisitWith};
+
+use crate::{
+    collect::Collect, strip_import::StripImport, strip_type::StripType, transform::Transform,
+    TsImportExportAssignConfig,
+};
+
+#[derive(Debug, Default, Serialize, Deserialize)]
+pub struct Config {
+    #[serde(default)]
+    pub verbatim_module_syntax: bool,
+
+    #[serde(default)]
+    pub is_tsx: bool,
+
+    /// Don't create `export {}`.
+    /// By default, strip creates `export {}` for modules to preserve module
+    /// context.
+    ///
+    /// https://github.com/swc-project/swc/issues/1698
+    #[serde(default)]
+    pub no_empty_export: bool,
+
+    /// Note: this pass handle jsx directives in comments
+    #[serde(default)]
+    pub pragma: Option<String>,
+
+    /// Note: this pass handle jsx directives in comments
+    #[serde(default)]
+    pub pragma_frag: Option<String>,
+
+    #[serde(default)]
+    pub import_export_assign_config: TsImportExportAssignConfig,
+}
+
+pub fn typescript(config: Config) -> impl Fold + VisitMut {
+    as_folder(TypeScript { config })
+}
+
+pub(crate) struct TypeScript {
+    pub config: Config,
+}
+
+impl VisitMut for TypeScript {
+    // TODO: program will be skipped in test_fixture
```

Comment:
`visit_mut_program` is skipped in  `test_fixture`. Is this expected behavior? @kdy1

Reply:
No, can you fix it too?